cli/src/semgrep/lsp/server.py

```diff
@@ -70,130 +62,11 @@ def poll_ls(self) -> None:
             return
         status = self.core_process.poll()
         if status is not None and status != 0:
-            self.notify_show_message(
-                1,
-                f"OSemgrep process died with status {status}, exiting...",
-            )
             self.exit_code = status
             self.polling = False
             self.stop()
 
-    def notify(self, method: str, params: JsonObject) -> None:
```

Comment:
just confirming my understanding, you're deleting all this because it's fully implemented in `osemgrep` now?

Reply:
What would it take to simply exec `osemgrep lsp` directly instead of shelling out to it?

Reply:
Yep.
Nothing, that's what we're doing now when semgrep lsp is called. The only time this code is run is for e2e tests. Once brandon has ported those we can remove this entire file and the e2e test file completely